optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -300,6 +307,12 @@ def forward(
             last_hidden_state = output_buffers["last_hidden_state"].view(output_shapes["last_hidden_state"])
         else:
             onnx_inputs = {"input_features": input_features.cpu().detach().numpy()}
+            if "attention_mask" in self.input_names:
+                onnx_inputs["attention_mask"] = attention_mask.cpu().detach().numpy()
+
+                # TODO: Replace with a better solution
+                if self.session.get_inputs()[1].type == "tensor(int64)":
+                    onnx_inputs["attention_mask"] = onnx_inputs["attention_mask"].astype(np.int64)
```

Comment:
Why is this needed?

Reply:
Some of the model tokenisers generate input in `int32` but the model export always happens with `int64` dtype (int input generator always generates this input).  This cause an error during inference because of incorrect datatype.

Reply:
Could you add a comment for this?

Reply:
Added